scripts/components/simple-horizontal-single-select/simple-horizontal-single-select.js

```diff
@@ -0,0 +1,178 @@
+ * @param {array<{label, value}>?} [props.optionLabels]                             - If passed, these labels/icons will be used instead the ones provided with `options`. Must be passed when `options` contain just values.
+ * @param {boolean} [props.iconOnly=false]                                          - If `true`, the buttons will only contain icons. If a label is also passed, it will be used for the button tooltip.
+ * @param {boolean} [props.allowWrap=true]                                          - If `true` and there is more options then can fit, the buttons will wrap to the row below.
+ * @param {boolean} [props.disabled=false]                                          - If `true`, the component will be disabled.
+ * @param {'none'|'inset'|'offset'} [props.border='none']                           - Sets the appearance of a border around the buttons.
+ * @param {'default'|'stretch'|'left'|'center'|'right'} [props.alignment='default'] - If `true` and there is more options then can fit, the buttons will wrap to the row below.
+ * @param {boolean} [props.includeWpBottomSpacing=false]                            - If `true`, the component will add bottom spacing to match other Gutenberg components.
+ * @param {boolean} [props.compactButtons=false]                                    - If `true`, the buttons are rendered smaller
+ */
+export const SimpleHorizontalSingleSelect = (props) => {
+	const {
+		help,
+		value,
+		label,
+		options,
+		onChange,
+		inlineLabel,
+		optionLabels,
+		iconOnly = false,
+		allowWrap = true,
+		disabled = false,
+		border = 'none',
+		alignment = 'default',
+		includeWpBottomSpacing = true,
+		compactButtons = false,
+	} = props;
+
+	const spacingConfig = {
+		default: {
+			width: 'max-content',
+		},
+		left: {
+			marginRight: 'auto',
+			justifyContent: 'flex-start',
+			width: 'max-content',
+		},
+		center: {
+			marginLeft: 'auto',
+			marginRight: 'auto',
+			justifyContent: 'center',
+			width: 'max-content',
+		},
+		right: {
+			marginLeft: 'auto',
+			justifyContent: 'flex-end',
+			width: 'max-content',
+		},
+		stretch: {
+			justifyContent: 'space-between',
+			width: '100%',
+		}
+	};
+
+	const borderConfig = {
+		offset: {
+			borderRadius: 3,
+			padding: '0.125rem',
+			border: '1px solid #ccc',
+		}
+	};
+
+	const control = (
+		<div
+			className={classnames([
+				'es-simple-h-single-select',
+				border === 'inset' ? 'has-inset-border' : '',
+			])}
+			style={{
+				display: 'flex',
+				flexWrap: allowWrap ? 'wrap' : 'nowrap',
+				maxWidth: '100%',
+				...(spacingConfig[alignment] ?? {}),
+				...(borderConfig[border] ?? {}),
+			}}
+		>
+			{options.map((item, i) => {
+				const icon = icons?.[item?.icon] ?? item?.icon ?? optionLabels?.[i]?.icon;
+				const label = item?.label ?? optionLabels?.[i]?.label;
+				const current = optionLabels ? item : item?.value;
+				const tooltip = item?.tooltip ?? optionLabels?.[i]?.tooltip ?? item?.label ?? optionLabels?.[i]?.label;
```

Comment:
![wtf](https://user-images.githubusercontent.com/8638515/159239822-2a25eed9-2699-46de-8620-61d90d31a71d.png)

Reply:
Let me break this down
```js
const icon = icons?.[item?.icon] ?? item?.icon ?? optionLabels?.[i]?.icon;
```
`icons?.[item?.icon]` - if string icon name is passed on, try it first with ES icons
`item?.icon` - if no such icon exists, try with just that element (or if string a dashicon)
`optionLabels?.[i]?.icon` - fallback if `optionLabels` is used to set the labels

Reply:
I know it looks worse, that's why I approved the PR 😂. Still, the one-liner is super hard to read 🙈